topiary-cli/src/main.rs

```diff
@@ -158,16 +197,35 @@ async fn run() -> CLIResult<()> {
         }
     };
 
-    formatter(
-        &mut input,
-        &mut output,
-        &query,
-        language,
-        &grammar,
-        operation,
-    )?;
-
-    output.into_inner()?.persist()?;
+    let tasks: Vec<_> = fmt_args
+        .into_iter()
+        .map(|(input, output, language, grammar, query)| {
+            tokio::spawn(async move {
+                let mut input: Box<dyn Read> = match input.as_str() {
+                    "-" => Box::new(stdin()),
+                    file => Box::new(BufReader::new(File::open(file).unwrap())),
+                };
+                let mut output: BufWriter<OutputFile> =
+                    BufWriter::new(OutputFile::new(&output).unwrap());
+
+                formatter(
+                    &mut input,
+                    &mut output,
+                    &query,
+                    &language,
+                    &grammar,
```

Comment:
Do we even need `grammar` as an argument? It's just derived from `language`.

Reply:
No we don't but I would like to discuss this real quick with you when we meet next.

Reply:
For documentation; we decided to leave this until we decided to refactor the API.